From a0e7001337b64a32eb0c9299385bfdd619c09373 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:33 -0400
Subject: [powerpc] Fix warnings for arch/powerpc/mm/numa.c

Message-id: <1410545655-205645-165-git-send-email-dzickus@redhat.com>
Patchwork-id: 94541
O-Subject: [RHEL7 PATCH 164/626] powerpc: Fix warnings for arch/powerpc/mm/numa.c
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit ec32dd663da29d0e2e3ca964d6a94c683a6582a9
Author: Robert Jennings <rcj@linux.vnet.ibm.com>
Date:   Mon Oct 28 09:20:50 2013 -0500

    powerpc: Fix warnings for arch/powerpc/mm/numa.c

    Simple fixes for sparse warnings in this file.

    Resolves:
    arch/powerpc/mm/numa.c:198:24:
            warning: Using plain integer as NULL pointer

    arch/powerpc/mm/numa.c:1157:5:
           warning: symbol 'hot_add_node_scn_to_nid' was not declared.
                    Should it be static?

    arch/powerpc/mm/numa.c:1238:28:
           warning: Using plain integer as NULL pointer

    arch/powerpc/mm/numa.c:1538:6:
           warning: symbol 'topology_schedule_update' was not declared.
                    Should it be static?

    Signed-off-by: Robert C Jennings <rcj@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/mm/numa.c b/arch/powerpc/mm/numa.c
index c447780..8e01af2 100644
--- a/arch/powerpc/mm/numa.c
+++ b/arch/powerpc/mm/numa.c
@@ -210,7 +210,7 @@ static const __be32 *of_get_usable_memory(struct device_node *memory)
  u32 len;
  prop = of_get_property(memory, "linux,drconf-usable-memory", &len);
  if (!prop || len < sizeof(unsigned int))
-  return 0;
+  return NULL;
  return prop;
 }
 
@@ -1206,7 +1206,7 @@ static int hot_add_drconf_scn_to_nid(struct device_node *memory,
  * represented in the device tree as a node (i.e. memory@XXXX) for
  * each memblock.
  */
-int hot_add_node_scn_to_nid(unsigned long scn_addr)
+static int hot_add_node_scn_to_nid(unsigned long scn_addr)
 {
  struct device_node *memory;
  int nid = -1;
@@ -1287,7 +1287,7 @@ static u64 hot_add_drconf_memory_max(void)
         struct device_node *memory = NULL;
         unsigned int drconf_cell_cnt = 0;
         u64 lmb_size = 0;
- const __be32 *dm = 0;
+ const __be32 *dm = NULL;
 
         memory = of_find_node_by_path("/ibm,dynamic-reconfiguration-memory");
         if (memory) {
@@ -1637,7 +1637,7 @@ static void topology_work_fn(struct work_struct *work)
 }
 static DECLARE_WORK(topology_work, topology_work_fn);
 
-void topology_schedule_update(void)
+static void topology_schedule_update(void)
 {
  schedule_work(&topology_work);
 }
-- 
1.7.1